
# --- ROCK-COPYRIGHT-NOTE-BEGIN ---
# 
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# Please add additional copyright information _after_ the line containing
# the ROCK-COPYRIGHT-NOTE-END tag. Otherwise it might get removed by
# the ./scripts/Create-CopyPatch script. Do not edit this copyright text!
# 
# ROCK Linux: rock-src/package/x11/xfree86/twm-defaults.patch
# ROCK Linux is Copyright (C) 1998 - 2004 Clifford Wolf
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# 
# --- ROCK-COPYRIGHT-NOTE-END ---

Some twm configs Clifford likes more ...

--- xc/programs/twm/system.twmrc.orig	2003-08-28 02:06:28.000000000 +0200
+++ xc/programs/twm/system.twmrc	2003-08-28 02:36:46.000000000 +0200
@@ -12,6 +12,8 @@
 # $XFree86: xc/programs/twm/system.twmrc,v 1.5 2003/08/04 10:32:30 eich Exp $
 #
 
+NoDefaults
+
 NoGrabServer
 RestartPreviousState
 DecorateTransients
@@ -21,6 +23,9 @@
 IconFont "-adobe-helvetica-bold-r-normal--*-100-*-*-*-*-*-*"
 IconManagerFont "-adobe-helvetica-bold-r-normal--*-100-*-*-*"
 #ClientBorderWidth
+RandomPlacement
+AutoRelativeResize
+IconifyByUnmapping
 
 Color
 {
@@ -47,48 +52,65 @@
 MoveDelta 3
 Function "move-or-lower" { f.move f.deltastop f.lower }
 Function "move-or-raise" { f.move f.deltastop f.raise }
-Function "move-or-iconify" { f.move f.deltastop f.iconify }
+Function "xiconify" { f.iconify f.showiconmgr }
 
 #
 # Set some useful bindings.  Sort of uwm-ish, sort of simple-button-ish
 #
-Button1 = : root : f.menu "defops"
 
-Button1 = m : window|icon : f.function "move-or-lower"
-Button2 = m : window|icon : f.iconify
-Button3 = m : window|icon : f.function "move-or-raise"
+LeftTitleButton "opendotMask" = f.function "xiconify"
+RightTitleButton "star" = f.delete
+
+Button1 = : root : f.menu "defops"
+Button2 = : root : f.menu "apps"
+Button3 = : root : f.exec "xterm &"
 
 Button1 = : title : f.function "move-or-raise"
-Button2 = : title : f.raiselower
+Button2 = : title : f.fullzoom
+Button3 = : title : f.function "move-or-lower"
 
-Button1 = : icon : f.function "move-or-iconify"
-Button2 = : icon : f.iconify
+Button1 = : icon : f.function "xiconify"
+Button3 = : icon : f.identify
+
+Button1 = : iconmgr : f.function "xiconify"
+Button2 = : iconmgr : f.raise
+Button3 = : iconmgr : f.identify
+
+Button1 = : frame : f.resize
+Button2 = : frame : f.raiselower
+Button3 = : frame : f.function "move-or-raise"
 
-Button1 = : iconmgr : f.iconify
-Button2 = : iconmgr : f.iconify
 
 #
 # And a menus with the usual things
 #
 menu "defops"
 {
-"Twm"	f.title
-"Iconify"	f.iconify
-"Resize"	f.resize
-"Move"		f.move
-"Raise"		f.raise
-"Lower"		f.lower
-""		f.nop
-"Focus"		f.focus
-"Unfocus"	f.unfocus
-"Show Iconmgr"	f.showiconmgr
-"Hide Iconmgr"	f.hideiconmgr
-""		f.nop
-"Xterm"		f.exec "exec xterm &"
-""		f.nop
-"Kill"		f.destroy
-"Delete"	f.delete
-""		f.nop
-"Restart"	f.restart
-"Exit"		f.quit
+	"TWM"		f.title
+	"Iconify"	f.iconify
+	"Resize"	f.resize
+	"Move"		f.move
+	"Raise"		f.raise
+	"Lower"		f.lower
+	"Zoom"		f.fullzoom
+	""		f.nop
+	"Focus"		f.focus
+	"Unfocus"	f.unfocus
+	"Show Iconmgr"	f.showiconmgr
+	"Hide Iconmgr"	f.hideiconmgr
+	""		f.nop
+	"Kill"		f.destroy
+	"Delete"	f.delete
+	""		f.nop
+	"Restart"	f.restart
+	"Exit"		f.quit
+}
+
+menu "apps"
+{
+	"Applications"		f.title
+	"xTerm"			f.exec "xterm &"
+	"xTerm (black)"		f.exec "xterm -fg white -bg black &"
+	"Mozilla"		f.exec "mozilla &"
+	"Icon Manager"		f.showiconmgr
 }
